### DISCUSSION TITLE
Is it possible to upgrade RKE2 using a different strategy

### DISCUSSION THREAD

**Discussion by athulg93** [rancher/rke2#6425] [Q&A]
So this is an upgrade requirement that we are looking into..
**Context:**
We install a custom OS with the following file partitions
/root
/var
/root1
/var1

The first OS is installed onto /root and /var. We have installed RKE2 and have a product running in this.

**Requirement**
When we upgrade the OS, we basically install the new version into /root1 & /var1 and copy the things we need into these 2 partitions..
The question is, if we directly copy the contents of /var/lib (which is where the RKE2 folders are present) into /var1/lib, will it work right away or will it fail?
Is RKE2 even designed to work like this?

**Comment by brandond** [comment]
It's certainly not something we've tested. If you make sure to stop rke2 and all the pods first it may work, but you'll have to validate this yourself. Have you considered adding a dedicated application partition for /var/lib/rancher?

**Reply by athulg93** [reply]
Thats what we are planning to do now.
use the kickstart file of the OS (we are using Rocky 9) and attach a partition for 
/var/lib/rancher
/var/lib/longhorn

Any other files that you recommend for RKE2 and rancher?

**Reply by brandond** [reply]
I can't speak to rancher or LH, but /var/lib/rancher and /etc/rancher/ should suffice for RKE2. Anything else will get recreated if it does not exist - although you will of course lose any existing pod state.
